gap/PolygonalComplexes/graphs.gi

```diff
@@ -698,3 +698,479 @@ InstallMethod( OnEdgeFacePaths,
     end
 );
 
+#######################################
+##
+##      All Surfaces Of A Graph
+##
+
+BindGlobal( "__SIMPLICIAL_EdgesFromAdjacencyMat",
+	function(mat)
+		local edges, i, j;
+
+        edges := Set([]);
+        for j in [1 .. Length(mat)] do
+			edges := Union(edges, List(ListBlist([1..j],mat[j]), i-> [j,i] ));
+        od;
+
+		return edges;
+end);
+
+
+# The following functions computes all cycles of a face graph <dig>
```

Comment:
Maybe you write something on how your method finds the cycles (maybe backtrack or something more efficient perhaps?)